test/jdk/java/awt/Frame/MaximizeUndecoratedTest.java

```
@@ -0,0 +1,129 @@
+                frame.setExtendedState(Frame.MAXIMIZED_BOTH);
+            });
+            robot.waitForIdle();
+            robot.delay(500);
+
+            // Colors sampled at top-left, top-right, bottom-right & bottom-left
+            // corners of maximized frame.
+            Point[] points = new Point[] {
+                    new Point(maxBounds.x + OFFSET, maxBounds.y + OFFSET),
+                    new Point(maxBounds.width - OFFSET, maxBounds.y + OFFSET),
+                    new Point(maxBounds.width - OFFSET, maxBounds.height - OFFSET),
+                    new Point(maxBounds.x + OFFSET, maxBounds.height - OFFSET)
+            };
+
+            if (!Stream.of(points)
+                       .map(p -> robot.getPixelColor(p.x, p.y))
+                       .allMatch(c -> c.equals(Color.GREEN))) {
+                saveScreenCapture();
+                throw new RuntimeException("Test Failed !! Frame not maximized.");
+            }
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.setExtendedState(Frame.NORMAL);
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void createUI() {
+        frame = new Frame("Test Maximization of Frame");
+        frame.setSize(SIZE, SIZE);
+        frame.setBackground(Color.GREEN);
+        frame.setResizable(true);
+        frame.setUndecorated(true);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void saveScreenCapture() {
+        BufferedImage image = robot.createScreenCapture(new Rectangle(new Point(),
+                                                                      screenSize));
+        try {
+            ImageIO.write(image, "png", new File("MaximizedFrame.png"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Is it best to just print the stack trace when a screenshot fails? Or is it better to throw a new runtime error?

---

It is better to re-throw it as RuntimeException. Updated.

---

> It is better to re-throw it as RuntimeException. Updated.

Not really…

If saving the image fails, we want to preserve *the original error message* which wasn't even thrown by this time. So printing stack trace in this particular is quite safe, the exception could be ignored completely.

---

Good point. Updated to print stacktrace in case IOException occurs to prevent masking the original more priority test exception.